h2o-r/h2o-package/R/connection.R

```diff
@@ -533,6 +531,12 @@ h2o.clusterStatus <- function() {
       }
     )
 
+  # NOTE for developers: keep the following blacklist in logically consistent with whitelist in java code - see water.H2O.checkUnsupportedJava, near line 1849
+  if (grepl("^java version \"1\\.[1-6]\\.", jver[1])) {
```

Comment:
What if the command produces an empty output, jver[1] will be an invalid reference - better to be safe

is there any harm in using `any(grepl("...", jver)`? just like the others? or check that the first line is really available in the output?

also I would move it after the other checks (they are more informative) and I would recommend the same thing as they do:

```
         "Please download the latest Java SE JDK 8 from the following URL:\n",
          "http://www.oracle.com/technetwork/java/javase/downloads/jdk8-downloads-2133151.html")
```

There is no reason why your case should be different, IMHO.

I think adding a unit tests would be a good idea. You can turn lines 534-549 into a function `.check.java.version(jver)` and unit test this function.

Reply:
For empty output and moving after other checks - totally correct, I will fix this.

Also you are right that this check should not provide different output. But, are you sure that we want to hard-code in so many places the exact JDK url, instead of pointing the user to more detailed and h2o-specific information about JDK, in all these cases?
To maintain the same level of user comfort, we can put that recommended url to that doc page, WDIT?